src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift

```diff
@@ -25,53 +25,104 @@ final class ExposureDetection {
 	private var completion: Completion?
 	private var progress: Progress?
 
+	private var countryKeypackageDownloader: CountryKeypackageDownloading
+
 	// MARK: Creating a Transaction
-	init(delegate: ExposureDetectionDelegate) {
+	init(
+		delegate: ExposureDetectionDelegate,
+		countryKeypackageDownloader: CountryKeypackageDownloading? = nil
+	) {
 		self.delegate = delegate
+
+		if let countryKeypackageDownloader = countryKeypackageDownloader {
+			self.countryKeypackageDownloader = countryKeypackageDownloader
+		} else {
+			self.countryKeypackageDownloader = CountryKeypackageDownloader(delegate: delegate)
+		}
 	}
 
 	func cancel() {
 		progress?.cancel()
 	}
 
-	// MARK: Starting the Transaction
-	// Called right after the transaction knows which data is available remotly.
-	private func downloadDeltaUsingAvailableRemoteData(_ remote: DaysAndHours?) {
-		guard let remote = remote else {
-			endPrematurely(reason: .noDaysAndHours)
-			return
-		}
-		guard let delta = delegate?.exposureDetection(self, downloadDeltaFor: remote) else {
-			endPrematurely(reason: .noDaysAndHours)
-			return
-		}
-		delegate?.exposureDetection(self, downloadAndStore: delta) { [weak self] error in
+	private func getSupportedCountries(completion: @escaping ([Country]) -> Void) {
+		delegate?.exposureDetection(supportedCountries: { [weak self] result in
 			guard let self = self else { return }
-			if error != nil {
-				self.endPrematurely(reason: .noDaysAndHours)
-				return
+
+			switch result {
+			case .success(let supportedCountries):
+				completion(supportedCountries)
+			case.failure:
+				self.endPrematurely(reason: .noSupportedCountries)
 			}
-			self.delegate?.exposureDetection(self, downloadConfiguration: self.useConfiguration)
-		}
+		})
+	}
+
+	private func getCountriesToDetect(supportedCountries: [Country]) -> [Country.ID] {
+		var countryIDs = supportedCountries.map { $0.id }
+		countryIDs.append(Country.defaultCountry().id)
+		return countryIDs
```

Comment:
would it make sense to eliminate possible duplicates here? its unlikely but if we get a dup here (e.g. the default country) we might run into unexpected trouble somewhere else.

Reply:
Good idea. Used a Set to do this.